### PR TITLE
Dependency updates: Spring Boot 3.0.4, Spring Framework 6.0.6, Spring Data Elasticsearch 5.0.3

### PR DESCRIPTION
**Pull request by reta**
### Description
Dependency updates: Spring Boot 3.0.3, Spring Framework 6.0.6, Spring Data Elasticsearch 5.0.3

### Issues Resolved
N/A

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@dlvenable @dblock could you take a look folks please

**Comment by reta** [comment]
@dlvenable @dblock  could you take a look folks please?

File: settings.gradle.kts [hunk 1]
```diff
@@ -63,7 +63,7 @@ pluginManagement {
       url = uri("https://plugins.gradle.org/m2/")
     }
     maven {
-      url = uri("https://repo.spring.io/milestone/")
+      url = uri("https://repo.spring.io/release/")
```
Comment (dlvenable) [inline settings.gradle.kts:66]: Can we remove this in favor of Maven Central?

**Review by dlvenable — APPROVED** [review]

File: settings.gradle.kts (hunk 1 above)
Comment (reta) [reply settings.gradle.kts:66]: That's the Gradle plugin repos, for all other artifacts we use Central

**Pull request merged**